diff --git a/doom-modeline-core.el b/doom-modeline-core.el
index bf51205..fe2e538 100644
--- a/doom-modeline-core.el
+++ b/doom-modeline-core.el
@@ -797,8 +797,8 @@ Throws an error if it doesn't exist."
 If DEFAULT is non-nil, set the default mode-line for all buffers."
   (when-let ((modeline (doom-modeline key)))
     (setf (if default
-              (default-value 'mode-line-format)
-            (buffer-local-value 'mode-line-format (current-buffer)))
+              (default-value 'header-line-format)
+            (buffer-local-value 'header-line-format (current-buffer)))
           (list "%e" modeline))))
 
 
