diff --git a/evil-commands.el b/evil-commands.el
index dabbf68..52a164e 100644
--- a/evil-commands.el
+++ b/evil-commands.el
@@ -37,6 +37,9 @@
 
 (declare-function imenu--in-alist "imenu")
 
+(autoload 'link-hint--get-link-at-point "link-hint")
+(autoload 'link-hint-open-link-at-point "link-hint")
+
 ;;; Motions
 
 ;; Movement commands, or motions, are defined with the macro
@@ -848,6 +851,8 @@ Calls `evil-lookup-func'."
          (doc    (get-char-property (point) 'widget-doc))
          (widget (or field button doc)))
     (cond
+     ((link-hint--get-link-at-point)
+      (link-hint-open-link-at-point))
      ((and widget
            (fboundp 'widget-type)
            (fboundp 'widget-button-press)
