diff --git a/ledger-report.el b/ledger-report.el
index eea11c97..17a2db7e 100644
--- a/ledger-report.el
+++ b/ledger-report.el
@@ -524,7 +524,9 @@ arguments returned by `ledger-report--compute-extra-args'."
     (unless ledger-report-use-header-line
       (insert (format "Report: %s\n" ledger-report-name)
               (format "Command: %s\n" clean-cmd)
-              (make-string (- (window-width) 1) ?=)
+              (if (featurep 'page-break-lines)
+                  ""
+                (make-string (- (window-width) 1) ?=))
               "\n\n"))
     (let* ((report (shell-command-to-string real-cmd)))
       (when ledger-report-use-native-highlighting
@@ -533,7 +535,9 @@ arguments returned by `ledger-report--compute-extra-args'."
         (insert report))
       (when (ledger-report--cmd-needs-links-p cmd)
         (save-excursion
-          (ledger-report--add-links))))))
+          (ledger-report--add-links)))
+      (when (featurep 'page-break-lines)
+        (page-break-lines--update-display-tables)))))
 
 (defun ledger-report-visit-source ()
   "Visit the transaction under point in the report window."
