diff --git a/lsp-mode.el b/lsp-mode.el
index 9fd49eb..7c2f348 100644
--- a/lsp-mode.el
+++ b/lsp-mode.el
@@ -1616,10 +1616,11 @@ WORKSPACE is the workspace that contains the diagnostics."
          (buffer (lsp--buffer-for-file file))
          (workspace-diagnostics (lsp--workspace-diagnostics workspace)))
 
-    (if (seq-empty-p diagnostics)
-        (remhash file workspace-diagnostics)
-      (when (or lsp-report-if-no-buffer buffer)
-        (puthash file (seq-map #'lsp--make-diag diagnostics) workspace-diagnostics)))
+    (when (hash-table-p workspace-diagnostics)
+      (if (seq-empty-p diagnostics)
+          (remhash file workspace-diagnostics)
+        (when (or lsp-report-if-no-buffer buffer)
+          (puthash file (seq-map #'lsp--make-diag diagnostics) workspace-diagnostics))))
 
     (run-hooks 'lsp-diagnostics-updated-hook)
 
