diff --git a/lisp/ibuffer.el b/lisp/ibuffer.el
index c9a7488..54a29eb 100644
--- a/lisp/ibuffer.el
+++ b/lisp/ibuffer.el
@@ -2195,18 +2195,7 @@ the value of point at the beginning of the line for that buffer."
 		  strname)))))
 	 (add-text-properties opos (point) '(ibuffer-title-header t))
 	 (insert "\n")
-	 ;; Add the underlines
-	 (let ((str (save-excursion
-		      (forward-line -1)
-		      (beginning-of-line)
-		      (buffer-substring (point) (line-end-position)))))
-	   (apply #'insert (mapcar
-			    (lambda (c)
-			      (if (not (or (eq c ?\s)
-					   (eq c ?\n)))
-				  ?-
-				?\s))
-			    str)))
+	 ""
 	 (insert "\n"))
        (point))
      `(ibuffer-title t font-lock-face ,ibuffer-title-face))
@@ -2332,7 +2321,7 @@ If optional arg SILENT is non-nil, do not display progress messages."
   (add-text-properties
    (point)
    (progn
-     (insert "[ " display-name " ]")
+     (insert display-name)
      (point))
    `(ibuffer-filter-group-name
      ,name
